File: Oc-Dome/Oc-Dome/Classes/Tools(工具)/HWCollectionViewController/HWNoDataView.m
//
//  HWNoDataView.m
//  kzyjsq
//
//  Created by 李含文 on 2018/12/25.
//  Copyright © 2018年 李含文. All rights reserved.
//

#import "HWNoDataView.h"

@interface HWNoDataView()
@property (weak, nonatomic) IBOutlet UIImageView *imageView;
@property (weak, nonatomic) IBOutlet UILabel *contentLabel;
@property (weak, nonatomic) IBOutlet UIButton *button;
/** <#注释#> */
@property(nonatomic, copy) void(^btnClickBlock)(void);
@end

@implementation HWNoDataView

+ (HWNoDataView *)noDataView {
    return [[NSBundle mainBundle] loadNibNamed:NSStringFromClass(self) owner:nil options:nil].lastObject;
}
- (void)awakeFromNib {
    [super awakeFromNib];
    self.button.hidden = YES;
    self.backgroundColor = [UIColor clearColor];
}
- (void)setImage:(UIImage *)image {
    _image = image;
    self.imageView.image = image;
}
- (void)setContent:(NSString *)content {
    _content = content;
    self.contentLabel.text = content;
}
- (void)setBtnTitle:(NSString *)title action:(void(^)(void))action {
    self.button.hidden = NO;
    [self.button setTitle:title forState:(UIControlStateNormal)];
    self.btnClickBlock = action;
}
- (IBAction)btnClick:(id)sender {
    if (self.btnClickBlock) {
        self.btnClickBlock();
    }
}
@end

File: Swift-Dome/LHWDome/Classes/Tool(工具类)/HWBaseCollectionView/HWNoDataView.swift
//
//  HWNoDataView.swift
//  WHYG
//
//  Created by Hanwen on 2018/5/27.
//  Copyright © 2018年 东莞市三心网络科技有限公司. All rights reserved.
//

import UIKit

class HWNoDataView: UIView {

    /// 缺失图片
    @IBOutlet weak var iconImageView: UIImageView!
    /// 缺失文字
    @IBOutlet weak var titleLabel: UILabel!
    /// 按钮 默认隐藏
    @IBOutlet weak var button: UIButton!
    /// 按钮点击回调
    var buttonClickBlcok:(()->())?
    class func noDataView(image:UIImage?, title:String?)->HWNoDataView{
        let nibView = Bundle.main.loadNibNamed("HWNoDataView", owner: nil, options: nil)
        if let view = nibView?.first as? HWNoDataView{
            view.iconImageView.image = image ?? hw_Image(named: "没数据缺省页")
            view.titleLabel.text = title ?? "暂无数据"
            return view
        }
        return HWNoDataView()
    }
    override func awakeFromNib() {
        super.awakeFromNib()
        button.addRounded()
        button.isHidden = true
        button.backgroundColor = HWMainColor()
    }
    @IBAction func buttonClick(_ btn: UIButton) {
        if buttonClickBlcok == nil {return}
        buttonClickBlcok!()
    }
    
    
}
